detector/vuls2/vendor.go

```diff
@@ -0,0 +1,240 @@
+	scanTypes "github.com/MaineK00n/vuls2/pkg/scan/types"
+	"github.com/future-architect/vuls/constant"
+	"github.com/future-architect/vuls/models"
+	rpm "github.com/knqyf263/go-rpm-version"
+)
+
+func resolveCveContentList(family string, sviX, sviY sourceVulnInfo, ccListX, ccListY []models.CveContent) (sourceVulnInfo, []models.CveContent) {
+	switch family {
+	// This logic is from old vuls's oval resolution. Maybe more widely applicable than these four distros.
+	case constant.RedHat, constant.CentOS, constant.Alma, constant.Rocky:
+		if sviX.rootID < sviY.rootID {
+			return sviY, ccListY
+		}
+		return sviX, ccListX
+	default:
+		return sviX, ccListX
+	}
+}
+
+func resolvePackageByRootID(family string, rootIDX, rootIDY dataTypes.RootID, statusX, statusY models.PackageFixStatus) models.PackageFixStatus {
+	switch family {
+	// This logic is from old vuls's oval resolution. Maybe more widely applicable than these four distros.
+	case constant.RedHat, constant.CentOS, constant.Alma, constant.Rocky:
+		former, latter := statusX, statusY
+		if rootIDY < rootIDX {
+			former, latter = statusY, statusX
+		}
+
+		fixedIn := statusX.FixedIn
+		if rpm.NewVersion(statusX.FixedIn).Compare(rpm.NewVersion(statusY.FixedIn)) < 0 {
+			fixedIn = statusY.FixedIn
+		}
+		if latter.NotFixedYet {
+			former.NotFixedYet = true
+			former.FixedIn = fixedIn
+			return former
+		}
+		latter.FixedIn = fixedIn
+		return latter
+	default:
+		return statusX
+	}
+}
+
+func mostPreferredTag(family string, segments []segmentTypes.Segment) segmentTypes.DetectionTag {
+	if len(segments) == 0 {
+		return segmentTypes.DetectionTag("")
+	}
+
+	s := slices.MaxFunc(segments, func(x, y segmentTypes.Segment) int {
+		return tagPreference(family, string(x.Tag)) - tagPreference(family, string(y.Tag))
+	})
+	return s.Tag
+}
+
+func resolvePackageByTag(family string, x, y pack) pack {
+	if tagPreference(family, string(x.tag)) < tagPreference(family, string(y.tag)) {
+		return y
+	}
+	return x
+}
+
+func resolveAdvisoryByTag(family string, x, y taggedAdvisory) taggedAdvisory {
+	if tagPreference(family, string(x.tag)) < tagPreference(family, string(y.tag)) {
+		return y
+	}
+	return x
+}
+
+func tagPreference(family string, tag string) int {
+	switch family {
+	case constant.RedHat, constant.CentOS:
+		switch {
+		case strings.HasSuffix(tag, "-including-unpatched"):
+			return 4
+		case strings.HasSuffix(tag, "-extras-including-unpatched"):
+			return 3
+		case strings.HasSuffix(tag, "-supplementary"):
+			return 2
+		default:
+			return 1
+		}
+	default:
+		return 1
+	}
+}
+
+func ignoresVulnerability(family string, v vulnerabilityTypes.Vulnerability, a *advisoryTypes.Advisory) bool {
+	switch family {
+	case constant.RedHat, constant.CentOS:
+		if a == nil && strings.Contains(v.Content.Description, "** REJECT **") {
+			return true
+		}
+		if a != nil && strings.Contains(a.Content.Description, "** REJECT **") {
+			return true
+		}
```

Comment:
Does it need to take into account the case where a ! = nil and v.Content.Description contains `** REJECT **`?

Reply:
done at https://github.com/future-architect/vuls/pull/2106/commits/532998318445cf1326bad0befb6ab98b3191cd00#diff-1be336d4b7528308cbd3a4c85b9bdb0938f40b2c973f26011f7048bd7352a7cdR109